controllers/haproxyloadbalancer_controller.go

```diff
@@ -369,25 +368,118 @@ func (r haproxylbReconciler) BackEndpointsForCluster(ctx *context.HAProxyLoadBal
 			}
 		}
 
-		machineEndpoints := make([]corev1.EndpointAddress, 0)
-		for i, addr := range machine.Status.Addresses {
-			if addr.Type == clusterv1.MachineExternalIP {
-				// TODO(frapposelli): Remove this check once HAproxy fully supports IPv6 - issue #859
-				if utilnet.IsIPv6String(addr.Address) {
-					continue
-				}
-				endpoint := corev1.EndpointAddress{
-					NodeName: pointer.StringPtr(fmt.Sprintf("%s-%d", machine.Name, i)),
-					IP:       addr.Address,
-				}
-				machineEndpoints = append(machineEndpoints, endpoint)
-			}
-		}
+		machineEndpoints := getMachineEndpoints(*machine)
 		endpoints = append(endpoints, machineEndpoints...)
 	}
 	return endpoints, nil
 }
 
+// getMachineEndpoints gets the endpoints from machines
+func getMachineEndpoints(machine clusterv1.Machine) (machineEndpoints []corev1.EndpointAddress) {
+	for i, addr := range machine.Status.Addresses {
+		if addr.Type == clusterv1.MachineExternalIP {
+			// TODO(frapposelli): Remove this check once HAproxy fully supports IPv6 - issue #859
+			if utilnet.IsIPv6String(addr.Address) {
+				continue
+			}
+			endpoint := corev1.EndpointAddress{
+				NodeName: pointer.StringPtr(fmt.Sprintf("%s-%d", machine.Name, i)),
+				IP:       addr.Address,
+			}
+			machineEndpoints = append(machineEndpoints, endpoint)
+		}
+	}
+	return machineEndpoints
+}
+
+// extraServicesForCluster returns the extra services that the loadbalancer proxies to
+func (r haproxylbReconciler) extraServicesForCluster(ctx *context.HAProxyLoadBalancerContext) ([]haproxy.Frontend, error) {
```

Comment:
Slightly confused why this method is returning just frontends as opposed to both front and backends.

Reply:
Changed the struct name to be Service